File: UtilitiesVB/Windows Forms/OlvExtension.vb
Imports System.Runtime.CompilerServices
Imports BrightIdeasSoftware

Public Module OlvExtension
    <Extension()>
    Public Sub AutoScaleColumnsToContainer(ByVal olv As ObjectListView)
        Dim containerwidth As Integer = olv.Width
        olv.BeginUpdate()
        Dim colswidth = 0
        For Each c As OLVColumn In olv.Columns
            colswidth += c.Width
        Next
        If colswidth <> 0 Then
            For Each c As OLVColumn In olv.Columns
                c.Width = CInt(Math.Round(c.Width * CDbl(containerwidth) / colswidth))
            Next
        End If
        olv.EndUpdate()
    End Sub
End Module

File: UtilitiesCS/HelperClasses/Windows Forms/OlvExtension.cs
using System;
using BrightIdeasSoftware;

namespace UtilitiesCS
{

    public static class OlvExtension
    {
        public static void AutoScaleColumnsToContainer(this ObjectListView olv)
        {
            int containerwidth = olv.Width;
            olv.BeginUpdate();
            int colswidth = 0;
            foreach (OLVColumn c in olv.Columns)
                colswidth += c.Width;
            if (colswidth != 0)
            {
                foreach (OLVColumn c in olv.Columns)
                    c.Width = (int)Math.Round(Math.Round(c.Width * (double)containerwidth / colswidth));
            }
            olv.EndUpdate();
        }
    }
}